distuv/f.go

```diff
@@ -0,0 +1,120 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+
+	"github.com/gonum/mathext"
+)
+
+// F implements the F-distribution, a two-parameter continuous distribution
+// with support over the positive real numbers.
+//
+// The F-distribution has density function
+//  sqrt(((d1*x)^d1) * d2^d2 / ((d1*x+d2)^(d1+d2))) / (x * B(d1/2,d2/2))
+// where B is the beta function.
+//
+// For more information, see https://en.wikipedia.org/wiki/F-distribution
+type F struct {
+	D1     float64 // Degrees of freedom for the numerator
+	D2     float64 // Degrees of freedom for the denominator
+	Source *rand.Rand
+}
+
+// CDF computes the value of the cumulative density function at x.
+func (f F) CDF(x float64) float64 {
+	return mathext.RegIncBeta(f.D1/2, f.D2/2, f.D1*x/(f.D1*x+f.D2))
+}
+
+// ExKurtosis returns the excess kurtosis of the distribution.
+//
+// ExKurtosis returns NaN if the D2 parameter is less or equal to 8.
+func (f F) ExKurtosis() float64 {
+	if f.D2 <= 8 {
+		return math.NaN()
+	}
+	num := 12 * (f.D1*(5*f.D2-22)*(f.D1+f.D2-2) + (f.D2-4)*(f.D2-2)*(f.D2-2))
```

Comment:
I don't know what typical parameters are of f distributions, but this looks badly conditioned to compute. I would split it up: first term is `((5*d2-22)/(d2-6)) * ((d1+d2-2)/(d2-8))` and similar for the second term. This should be much better behaved when d1 or d2 gets large, keeping each term O(1) rather than dividing two really large numbers.